### ISSUE TITLE
JSON.stringify should disallow BigInt value

### ISSUE THREAD

**Issue by wyattscarpenter** [microsoft/TypeScript#63333]
### ⚙ Compilation target

es2023

### ⚙ Library

lib.es5.d.ts

### Missing / Incorrect Definition

`JSON.stringify` is correctly marked with `@throws {TypeError} If a circular reference or a BigInt value is found.`, but it types its input `value` parameter as `any`. That type could be improved so that BigInt can't even be passed into the function (turning one runtime error into a compiletime error). 

Perhaps even every type that contains a BigInt can and should be excluded. I'm not very good at TypeScript but that might be possible.

Note that if a replacer function is specified, this complicates the situation somewhat.

This issue is distinct from the perennial proposal to correctly treat `undefined` in JSON.stringify (eg https://github.com/microsoft/TypeScript/issues/18879). Hopefully, the false-positive impact of this one should be much lower, and thus it can be accepted.

### Sample Code

```TypeScript
//These are runtime errors, but TypeScript doesn't complain.
//"Uncaught TypeError: Do not know how to serialize a BigInt"
JSON.stringify(1n);
JSON.stringify([1n]);
```

### Documentation Link

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Errors/BigInt_not_serializable

https://tc39.es/ecma262/multipage/structured-data.html#sec-json.stringify

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/JSON/stringify

**Comment by nmain** [comment]
What type would the argument to `JSON.stringify` have?  If it tried to list only things that were guaranteed to be JSON stringifyable, that would be https://github.com/microsoft/TypeScript/issues/1897.  If you wanted `Exclude<unknown, bigint>`, that would be https://github.com/microsoft/TypeScript/issues/4196.
